src/main.js

```diff
@@ -1,29 +1,51 @@
 //function that checks if the env value is valid
 //value cannot start with a digit and must only contain alphanumeric characters or an underscore
 const checkValidationOfValues = envSpecString => {
+  const validValues = ['color','date','datetime-local','email','month','number','password','tel','text','time','url','week','default'];
   const alphanumericThatDoesNotStartWithDigit = /^[A-Z_][0-9A-Z_]*$/;
-  let envValuesChecked = envSpecString.split("\n"); //split lines based on \n character
+  let envValArray = parseVarFromVal(envSpecString.split("\n")); //split lines based on \n character and parse them
 
-  return (envValuesChecked = envValuesChecked.filter(element => {
-    return element.match(alphanumericThatDoesNotStartWithDigit); //keep valid values
-  }));
+  return envValArray = envValArray.filter( element =>{ //check for valid variables AND values
+    if(element[0].match(alphanumericThatDoesNotStartWithDigit)&&(validValues.includes(element[1]))){
+      return element;
+    }
+  })
 };
 
+//function that separates the variables from their values and returns them as separated arrays
+const parseVarFromVal = envSpecAsArray => {
+  let envArrayParsed = [];
+  let parsPos = -1;
+  return envArrayParsed = envSpecAsArray.map( element =>{
+    parsPos = element.indexOf(":");
+    if(parsPos<0){
```

Comment:
We should simplify this. We can use the [`includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/includes) method, present on strings.